connexion/decorators/parameter.py

```diff
@@ -215,6 +223,86 @@ def get_arguments(
         # TRACE requests MUST NOT include a body (RFC7231 section 4.3.8)
         return ret
 
+    # Special handling for file uploads
+    body_schema = operation.body_schema(content_type)
+
+    # Check for different schema versions
+    is_openapi31 = (
+        hasattr(body_schema, "get") and body_schema.get("components", {}) is not None
+    )
+    is_swagger2 = isinstance(operation, Swagger2Operation)
+
+    # Check if request has multipart/form-data content type and contains a 'file'
+    is_file_upload = False
+    if content_type.startswith("multipart/form-data") and files and "file" in files:
+        is_file_upload = True
+
+    # Handle file uploads - make a copy to avoid modifying the original
+    if is_file_upload:
+        files = dict(files)
+
+    # Handling for OpenAPI file uploads
+    if is_file_upload and not is_swagger2:
+        # Preserve handlers using body vs form split
+        body_name = sanitize(operation.body_name(content_type))
+
+        # Handle all OpenAPI file upload cases
+        if is_file_upload and isinstance(body, dict):
```

Comment:
I think this use of `is_file_upload` can be safely dropped since you checked that at line 245